### PR TITLE
docs: add Discourse link to front page

### PR DESCRIPTION
**Pull request by s-makin**
```
Now that the docs are live, they need to be linked up
with the forums. I have therefore added a link to the
beta invitation post made by Lech on Discourse. 

I have also reformatted the "get help" section slightly.
Please let me know if you think I need to tweak the
wording, or if you can think of any other links we should
add while I'm at it! 
```

### CODE REVIEW

**Review by renanrodrigo — APPROVED** [review]

File: docs/index.rst [hunk 1]
```diff
@@ -108,3 +119,4 @@ constructive feedback.
 .. _Code of conduct: https://ubuntu.com/community/code-of-conduct
 .. _Contribute: https://github.com/canonical/ubuntu-advantage-client/blob/main/CONTRIBUTING.md
 .. _IRC channel on Libera: https://kiwiirc.com/nextclient/irc.libera.chat/ubuntu-server
+.. _our Discourse forum: https://discourse.ubuntu.com/t/ubuntu-pro-beta-invitation/30957
```
Comment (orndorffgrant) [inline docs/index.rst:122]: Alternatively, we could link to the "Ubuntu Pro" topic on discourse https://discourse.ubuntu.com/c/ubuntu-pro/116
Then people can create new posts or browse the existing posts.
What do you think?

Comment (s-makin) [reply docs/index.rst:122]: I did have a scan through all the posts on there, but it looks a bit...empty...so I went for the one that was best constructed and most helpful to new users.

That said, I can change it to the topic if you think that's better, or we could create a new post specifically to welcome peoples' feedback/questions/etc on the pro client.

Comment (renanrodrigo) [reply docs/index.rst:122]: hmmmm Grant is right, this one is for the beta. Pointing to the topic makes more sense? I would expect it to start empty as people didn't have the time to fill it yet (:

Comment (s-makin) [reply docs/index.rst:122]: Should we start a "welcome" post, in that case?

Comment (s-makin) [reply docs/index.rst]: On reflection, I think Grant was right in the first place - I've updated the link as he suggested to the one for the Pro Discourse main channel.

**Pull request merged**